### ISSUE TITLE
Flutter tool should collect C++ crashes in dev mode and offer to report them

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#22818]
Engine crashes should be rare.

Ideally we should catch engine crashes, and offer to report them.  Whether we catch them just using the flutter tool (to read logs) or we do something more like adding breakpad to the engine in debug mode, we could catch crashes (e.g. https://github.com/flutter/flutter/issues/19590) and the `flutter` tool could prompt the user to report them to our crash database.

It's possible we could automatically report them, but that would require more consideration as the stacks my include non-Flutter code.

Thoughts?

**Comment by eseidelGoogle** [comment]
Curious to your thoughts @gspencergoog or @cbracken.   It scares me a bit that we're blind to this kind of crasher.

I know @mraleph has also been adding crash-reporting to the `dart` command line executable, which could also help in this space.

**Comment by mklim** [comment]
From what I can tell adding generic C++ crash handling is also a prerequisite for #6500.

**Comment by cbracken** [comment]
Agreed; collecting these on an opt-in basis is likely to get us more data, with less manual bug-filing by users. Assuming we've got automatic triaging on the server side as we collect these, this also provides objective data on the frequency of these, which we only get from thumbs-ups on issues (assuming they were filed and easily searchable) today.
